go-controller/pkg/ovn/master.go

```diff
@@ -528,10 +580,8 @@ func (oc *Controller) addNode(node *kapi.Node) (hostsubnet *net.IPNet, err error
 	// Set the HostSubnet annotation on the node object to signal
 	// to nodes that their logical infrastructure is set up and they can
 	// proceed with their initialization
-	err = oc.kube.SetAnnotationOnNode(node, OvnHostSubnet, hostsubnet.String())
+	err = oc.addNodeAnnotations(node, hostsubnet.String(), false)
```

Comment:
The comment here seems to be wrong; the node knows when to proceed based on when its logical switch is created, not the annotation. Thus, also, there's no reason to pass `false` for `replace`, and therefore no reason to even have the `replace` arg.

Reply:
@danwinship you are right that the node proceeds upon the logical switch creation. However, I found an issue with that behavior (for which I will submit the fix very soon).

Say, that the master created the LogicalSwitch with ensureNodeLogicalNetwork() on the master. However, master failed to annotate the subnet information on the node. However, the node proceeded with using the LogicalSwitch's subnet. Now, when the master restarts it fails to account for that subnet since the corresponding node didn't have subnet annotation. It will assign the node with a new subnet.

I think the comment is correct, and we need to fix the node to wait on the annotation on the node since that is the last thing master does with regard to a node.

Reply:
> Thus, also, there's no reason to pass `false` for `replace`, and therefore no reason to even have the `replace` arg.

Are you saying to do this addNodeAnnotations() always ?
`nodeAnnotations[OvnHostSubnetLegacy] = nil` 

If so, I agree. We can remove that statement in the next release of OVN-K8s (because by then all the nodes must have been annotated with the new annotation name)